### DISCUSSION TITLE
Preference set to 'all' not working for all tenants

### DISCUSSION THREAD

**Discussion by pranavirabatti** [Sunbird-Lern/Community#20] [Issues] [answered]
Sunbird portal version: 4.4.1

I've created a preference with orgId set to 'all'. When I try to read any random tenant I see this error given below. 

Endpoint: /org/v2/preferences/create

```
{
    "id": "api.org.preferences.read",
    "ver": "v2",
    "ts": "2022-07-06 10:38:24:410+0000",
    "params": {
        "resmsgid": null,
        "msgid": "245beaf7e197c818e345b561609f21c4",
        "err": "PREFERENCE_NOT_FOUND",
        "status": "PREFERENCE_NOT_FOUND",
        "errmsg": "preference certRules not found in the org 0130171255884513283"
    },
    "responseCode": "RESOURCE_NOT_FOUND",
    "result": {}
}
```
@amit-tarento @HarishGangula please help me here

**Comment by pranavirabatti** [comment, marked as answer]
@amit-tarento @reshmi-nair please help me here

**Reply by reshmi-nair** [reply]
@pranavirabatti If tenant preference with orgId set to 'all', then while reading also orgId "all" has to be passed in request. The current read API, does not return the default value if preference for an org does not exist.

**Reply by pranavirabatti** [reply]
Ok. Thanks!
